packages/trace-viewer/src/traceModel.ts

```diff
@@ -238,16 +249,16 @@ export class TraceModel {
     }
   }
 
-  private _modernize(event: any): trace.TraceEvent | null {
+  private _modernize(event: any): trace.TraceEvent[] {
     if (this._version === undefined)
-      return event;
-    const lastVersion: trace.VERSION = 5;
+      return [event];
+    const lastVersion: trace.VERSION = 6;
     for (let version = this._version; version < lastVersion; ++version) {
       event = (this as any)[`_modernize_${version}_to_${version + 1}`].call(this, event);
       if (!event)
-        return null;
+        return [];
     }
-    return event;
+    return Array.isArray(event) ? event : [event];
```

Comment:
I guess we have to change all modernizers to return an array now, to avoid this hack.